src/openms/source/METADATA/ID/IdentificationData.cpp

```diff
@@ -190,42 +212,59 @@ namespace OpenMS
     return result.first;
   }
 
-
-  IdentificationData::DataQueryRef
-  IdentificationData::registerDataQuery(const DataQuery& query)
+  IdentificationData::ObservationRef
+  IdentificationData::registerObservation(const Observation& obs)
   {
-    // reference to spectrum or feature is required:
-    if (query.data_id.empty())
+    if (!no_checks_)
     {
-      String msg = "missing identifier in data query";
-      throw Exception::IllegalArgument(__FILE__, __LINE__,
-                                       OPENMS_PRETTY_FUNCTION, msg);
+      // reference to spectrum or feature is required:
+      if (obs.data_id.empty())
+      {
+        String msg = "missing identifier in observation";
+        throw Exception::IllegalArgument(__FILE__, __LINE__,
+                                         OPENMS_PRETTY_FUNCTION, msg);
+      }
+      // ref. to input file must be valid:
+      if (!isValidReference_(obs.input_file, input_files_))
+      {
+        String msg = "invalid reference to an input file - register that first";
+        throw Exception::IllegalArgument(__FILE__, __LINE__,
+                                         OPENMS_PRETTY_FUNCTION, msg);
+      }
     }
-    // ref. to input file may be missing, but must otherwise be valid:
-    if (query.input_file_opt && !isValidReference_(*query.input_file_opt,
-                                                   input_files_))
+
+    // can't use "insertIntoMultiIndex_" because Observation doesn't have the
+    // "steps_and_scores" member (from ScoredProcessingResult)
+    auto result = observations_.insert(obs);
+    if (!result.second) // existing element - merge in new information
     {
-      String msg = "invalid reference to an input file - register that first";
-      throw Exception::IllegalArgument(__FILE__, __LINE__,
-                                       OPENMS_PRETTY_FUNCTION, msg);
+      observations_.modify(result.first, [&obs](Observation& existing)
```

Comment:
can you document what information get's merged here?

Reply:
See https://github.com/OpenMS/OpenMS/pull/5352#discussion_r758288483.

Reply:
Then maybe just a generic comment like „set or overwrite values associated with current key“ ? If that is what it does …